pkgs/development/python-modules/pulsar-client/default.nix

```
@@ -1,133 +1,54 @@
 {
   lib,
-  stdenv,
   buildPythonPackage,
-  fetchPypi,
-  python,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  libpulsar,
+  pybind11,
+  certifi,
 }:
-let
-  version = "3.4.0";
 
-  inherit (python) pythonVersion;
+buildPythonPackage rec {
+  pname = "pulsar";
+  version = "3.7.0";
+  format = "setuptools";
```

```suggestion
  pyproject = true;
```
and set `build-system = [ setuptools ]`

---

Done